packages/openid4vc/src/openid4vc-issuer/router/credentialOfferEndpoint.ts

```diff
@@ -0,0 +1,89 @@
+import type { OpenId4VcIssuanceRequest } from './requestContext'
+import type { OpenId4VcIssuanceSessionStateChangedEvent } from '../OpenId4VcIssuerEvents'
+import type { Router, Response } from 'express'
+
+import { joinUriParts, EventEmitter } from '@credo-ts/core'
+
+import { getRequestContext, sendErrorResponse } from '../../shared/router'
+import { OpenId4VcIssuanceSessionState } from '../OpenId4VcIssuanceSessionState'
+import { OpenId4VcIssuerEvents } from '../OpenId4VcIssuerEvents'
+import { OpenId4VcIssuerModuleConfig } from '../OpenId4VcIssuerModuleConfig'
+import { OpenId4VcIssuanceSessionRepository } from '../repository'
+
+export interface OpenId4VciCredentialOfferEndpointConfig {
+  /**
+   * The path at which the credential offer should should be made available. Note that it will be
+   * hosted at a subpath to take into account multiple tenants and issuers.
+   *
+   * @default /offers
+   */
+  endpointPath: string
+}
+
+export function configureCredentialOfferEndpoint(router: Router, config: OpenId4VciCredentialOfferEndpointConfig) {
+  router.get(
+    joinUriParts(config.endpointPath, [':credentialOfferId']),
+    async (request: OpenId4VcIssuanceRequest, response: Response, next) => {
+      const { agentContext, issuer } = getRequestContext(request)
+
+      try {
+        const openId4VcIssuanceSessionRepository = agentContext.dependencyManager.resolve(
+          OpenId4VcIssuanceSessionRepository
+        )
+        const issuerConfig = agentContext.dependencyManager.resolve(OpenId4VcIssuerModuleConfig)
+
+        // TODO: is there a cleaner way to get the host (including port)?
+        const [, , host] = issuerConfig.baseUrl.split('/')
+
+        const credentialOfferUri = `${request.protocol}://${host}${request.originalUrl}`
+        const openId4VcIssuanceSession = await openId4VcIssuanceSessionRepository.findSingleByQuery(agentContext, {
+          issuerId: issuer.issuerId,
+          credentialOfferUri,
+        })
+
+        if (!openId4VcIssuanceSession || !openId4VcIssuanceSession.credentialOfferPayload) {
+          return sendErrorResponse(response, agentContext.config.logger, 404, 'not_found', 'Credential offer not found')
+        }
+
+        if (
+          ![OpenId4VcIssuanceSessionState.OfferCreated, OpenId4VcIssuanceSessionState.OfferUriRetrieved].includes(
+            openId4VcIssuanceSession.state
+          )
+        ) {
+          return sendErrorResponse(
```

Comment:
I can imagine we can simplify this function quite a bit. Again, not too relevant for this PR, but for improvements in the future.